seatunnel-api/src/main/java/org/apache/seatunnel/api/configuration/ConfigAdapterSpi.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.api.configuration;
+
+import java.nio.file.Path;
+import java.util.Map;
+
+/**
+ * Adapter mode to support convert other config to HOCON.
+ */
+public interface ConfigAdapterSpi {
+
+    /**
+     * Use the converter adapter with a file extension.
+     *
+     * @param fileExtension config file extension
+     * @return enable use this SPI
+     */
+    boolean checkFileExtension(String fileExtension);
```

Comment:
```suggestion
    String extensionIdentifier();
```
Use the identifier to verify this spi is nor not support extension is better. What do you think about it?

Reply:
Because an adapter may support multiple file extension identifiers,

If so, you need to return the array.

`String[] extensionIdentifier();`

And use for-each `org.apache.commons.lang3.StringUtils.equalsIgnoreCase()` to check in `ConfigAdapterUtils`.

@TyrantLucifer

Reply:
Yup, you are right. Let's go with your plan.